### DISCUSSION TITLE
How to close Terminal app current instance and launch a new instance

### DISCUSSION THREAD

**Discussion by rajendrasp** [microsoft/terminal#15422] [Q&A]
I have following scenario - 
I am adding a new Terminal profile and setting it as default using a Powershell script.
This script is running in current instance of Terminal app.
When this script is completed, I want to close the current instance of Terminal and launch a new instance so that new default profile gets loaded.
How can I do it?

I tried with exit in the script - it is not closing the Terminal.
I tried with Stop-Process $PID - it is not closing Terminal even though it is killing Powershell process.

Any idea how can I do this?

**Comment by zadjii-msft** [comment]
The Terminal _should_ "hot reload" it's settings. If you change the `settings.json` file, we'll immediately reload the settings. You shouldn't need to close and restart the window at all. Is that not working for you?

**Reply by zadjii-msft** [reply]
for example:
![c1ff42f3-a267-44fd-8395-18c9636199e4](https://github.com/microsoft/terminal/assets/18356694/0001d318-615e-4e8d-bc2c-2905c35ae118)

**Comment by rajendrasp** [comment]
Thanks @zadjii-msft Yes it is hot reloading for me. I guess, creating a new tab is not that much of a trouble. But I was looking for closing the current instance and launching a new one so that I start with new Starting dir and admin mode and does not have to keep an unnecessary tab/window open and user does not have to manually click to open a new tab.

Shouldn't Stop-Process work though? Something I am doing wrong there?
